validation/src/main/java/io/micronaut/validation/exceptions/ConstraintExceptionHandler.java

```diff
@@ -46,14 +48,19 @@
 @Requires(classes = {ConstraintViolationException.class, ExceptionHandler.class})
 public class ConstraintExceptionHandler implements ExceptionHandler<ConstraintViolationException, HttpResponse<JsonError>> {
 
+    @Inject
+    private JacksonConfiguration jacksonConfiguration;
```

Comment:
private field injection requires reflection, please change this to a field representing the configuration and use constructor injection and then add an additional constructor to maintain compatibility. Something like:

```java
private final boolean alwaysSerializeErrorsAsList;

@Inject
public ConstraintExceptionHandler(JacksonConfiguration jacksonConfiguration) {
       this. alwaysSerializeErrorsAsList = jacksonConfiguration.isAlwaysSerializeErrorsAsList();
}

public ConstraintExceptionHandler() {
      this.alwaysSerializeErrorsAsList = true
}
```